### DISCUSSION TITLE
Creating User after user registration confirmation: why UserRegistrationConfirmedNotification is not consumed directly by UserAccess?

### DISCUSSION THREAD

**Discussion by forgoty** [kgrzybek/modular-monolith-with-ddd#323] [Q&A]
I've noticed that single exception when modules are directly call each other is when [**Registrations** module calls **UserAccess**](https://github.com/kgrzybek/modular-monolith-with-ddd/blob/master/docs/C4/c3_components.puml#L44). This happens as the reaction on [**UserRegistrationConfirmedNotification**](https://github.com/kgrzybek/modular-monolith-with-ddd/blob/master/src/Modules/Registrations/Application/UserRegistrations/ConfirmUserRegistration/UserRegistrationConfirmedNotificationHandler.cs#L27). 

The question is: why this notification isn't consumed by **UserAccess** module itself?

**Comment by nefosfatus** [comment]
Same question. @kgrzybek Please explain why you made an exception in this case? Because registrations and user access are closely related?

**Comment by nefosfatus** [comment]
@forgoty I read [this article](https://www.kamilgrzybek.com/blog/posts/modular-monolith-integration-styles) and I think I understand it. There are several methods to communicate between modules, each with its own advantages:

Messaging - Ensures the lowest coupling, modularity, and autonomy but comes at the cost of complexity.
Direct Call - Provides lower coupling than Shared Database Data, encapsulates modules, and is relatively simple.
The author, suggests that "most often, when we are talking about modular architecture, it is best to use Direct Call and Messaging together." This shows that we can use the simple Direct Call method when needed.
